test/serializationtests.jl

```diff
@@ -0,0 +1,39 @@
+using Serialization
+@testset "Serialize and Deserialize" begin
+
+    cache = LRU{Int, Int}(maxsize=100_000)
+
+    # Populate the cache with dummy data
+    for i in 1:1_000_000
+        cache[i] = i+1
+    end
+    serialize("cache.jls", cache)
+    deserialized_cache = deserialize("cache.jls")
+    rm("cache.jls")
```

Comment:
We can avoid the filesystem by using an IOBuffer:
```suggestion
    io = IOBuffer()
    serialize(io, cache)
    seekstart(io)
    deserialized_cache = deserialize(io)
```

Reply:
Also a good change, thanks